### PR TITLE
identify: filter received addresses based on the node's remote address

### PR DESCRIPTION
**Pull request by marten-seemann**
Unfortunately, this is impossible to test. I tried using mocknet, but of course that thing is so horribly broken that it can't even reliably set the correct remote address of a connection. It fails at the most easy thing it was designed to do.

### CODE REVIEW

**Comment by sukunrt** [comment]
If you want a connection with a remote addr on private IP and a remote addr on loopback ip, it should be possible to do this by listening on "0.0.0.0" address and dialing only the private or the loopback address. Testing public ips does look impossible.

File: p2p/protocol/identify/id.go [hunk 1]
```diff
@@ -767,12 +767,11 @@ func (ids *idService) consumeMessage(mes *pb.Identify, c network.Conn, isPush bo
 	// add signed addrs if we have them and the peerstore supports them
 	cab, ok := peerstore.GetCertifiedAddrBook(ids.Host.Peerstore())
 	if ok && signedPeerRecord != nil {
-		_, addErr := cab.ConsumePeerRecord(signedPeerRecord, ttl)
-		if addErr != nil {
-			log.Debugf("error adding signed addrs to peerstore: %v", addErr)
+		if _, err := cab.ConsumePeerRecord(signedPeerRecord, ttl); err != nil {
+			log.Debugf("error adding signed addrs to peerstore: %v", err)
```
Comment (sukunrt) [inline p2p/protocol/identify/id.go:771]: for this to be effective, these addresses need to be filtered too, right?

Comment (marten-seemann) [reply p2p/protocol/identify/id.go:771]: Right. I've done some digging and convinced myself that the `ConsumePeerRecord` API is flawed. Will open a separate issue for that.

**Comment by marten-seemann** [comment]
@sukunrt Should we merge this PR, and do the fixes for signed peer records when we tackle #2304?

**Review by sukunrt — APPROVED** [review]

**Comment by sukunrt** [comment]
Yes. We can take signed peer records separately.

**Pull request merged**